lib/services/remove_watcher.rb

```diff
@@ -34,8 +34,12 @@ def initialize(work_package, user)
 
   def run(success: -> {}, failure: -> {})
     if @work_package.watcher_users.include?(@user)
+      @watcher = @work_package.watchers.find_by_user_id(@user.id)
```

Comment:
There is a duplication here now: 

`@watcher = @work_package.watchers.find_by_user_id(@user.id)`

could now also be used to check whether the user is a watcher.

Why is `@watcher` an instance variable btw? A local variable would suffice.